### PR TITLE
Allow overriding the default browser profile directory

### LINKED ISSUE #21: Firefox restore problem
Hey @gregpabian 

I've noticed the following thing when I close Firefox (both with instance.stop() and with process.kill()): the next time I open it it tells me to start in safe mode and restore my tabs.

I'm using Windows 7 and Node v0.10.x.

I'm trying to figure out how to solve this and will keep you updated if I find something.

Thanks!

### PR DESCRIPTION
**Pull request by pimterry**
I need this because the Firefox setup creates a temp folder every time. This fixed https://github.com/benderjs/browser-launcher2/issues/21, because repeatedly killing a firefox instance using a profile folder can corrupt it.

In my case though, I'm launching a Firefox instance that I expect the user to close manually most of the time, so this isn't necessary, and I'd like the profile to persist between runs, which this currently breaks. Being able to override the profile to provide a fixed directory solves this nicely for me, and doesn't change the existing behaviour.

As a bonus, rather than create a firefox-specific option, I've updated all the other setups too, so that you can now override the profile directory everywhere, if you want to for some reason.

### CODE REVIEW

**Comment by mitchhentges** [comment]
Hey, thanks for the PR!

I'll look into this a little more in the coming week, but my initial concern is adding options that are browser-specific - in this case, someone might be starting IE and want to set the `profile`, and are confused when the option doesn't affect the launch, since this option only affects Firefox and Chrome.

I'll investigate further soon, I'm not sure what a more-elegant solution would be just yet, or whether it would affect this PR

**Comment by pimterry** [comment]
Interesting, ok. Are there no other existing parameters that behave differently across browsers like this?

One option would be to detect this case, and explicitly throw an error if you try to pass a profile path to a browser that doesn't support it. How does that sound?

**Review by mitchhentges — APPROVED** [review]
Looks great, nice work!
Going to add @tomitm as a reviewer to get one more set of eyes, then I'll release after this is merged.

-----

Looks like at least `proxy` is another option that doesn't affect all browsers (e.g. IE isn't affected by the `proxy` option). So, though it's still an issue that it's hard to tell which options work against which browsers, this PR doesn't introduce this issue as a regression.

TL;DR :+1: :)

**Comment by pimterry** [comment]
@tomitm any thoughts on this? I'd love to get this merged.

**Review by tomitm — APPROVED** [review]

**Comment by mitchhentges** [comment]
Thanks @pimterry! I'll get this released in `v1.3.3` in a couple minutes here :smile:

**Comment by mitchhentges** [comment]
`v1.3.3` has been released :package:

**Comment by pimterry** [comment]
Superb, thanks!
